src/main/java/hudson/remoting/RemoteInvocationHandler.java

```diff
@@ -410,31 +418,40 @@ public void run() {
             inQueue.set(false); // we have started execution and running is true, so queued can be reset
             try {
                 long nextSweep = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(200);
+                int count = 0;
                 while (!referenceLists.isEmpty()) {
                     try {
-                        Reference<?> ref = queue.remove(100);
-                        if (ref instanceof PhantomReferenceImpl) {
-                            PhantomReferenceImpl r = (PhantomReferenceImpl) ref;
-                            final Channel.Ref channelRef = r.channel;
-                            try {
-                                r.cleanup();
-                            } catch (IOException e) {
-                                logger.log(Level.WARNING,
-                                        String.format("Couldn't clean up oid=%d from %s", r.oid, r.origin), e);
-                            } catch (Error e) {
-                                logger.log(Level.SEVERE,
-                                        String.format("Couldn't clean up oid=%d from %s", r.oid, r.origin), e);
-                                throw e; // pass on as there is nothing we can do with an error
-                            } catch (Throwable e) {
-                                logger.log(Level.WARNING,
-                                        String.format("Couldn't clean up oid=%d from %s", r.oid, r.origin), e);
-                            } finally {
-                                if (channelRef != null) {
-                                    final List<PhantomReferenceImpl> referenceList = referenceLists.get(channelRef);
-                                    if (referenceList != null) {
-                                        referenceList.remove(r);
-                                        if (channelRef.channel() == null) {
-                                            cleanList(referenceList);
+                        long drainUntil = System.currentTimeMillis() + 200;
+                        long remaining;
+                        while ((remaining = drainUntil - System.currentTimeMillis()) > 0) {
+                            Reference<?> ref = queue.remove(remaining);
+                            if (ref == null) {
+                                break;
+                            }
+                            if (ref instanceof PhantomReferenceImpl) {
+                                count++;
+                                PhantomReferenceImpl r = (PhantomReferenceImpl) ref;
+                                final Channel.Ref channelRef = r.channel;
+                                try {
+                                    r.cleanup();
+                                } catch (IOException e) {
+                                    logger.log(Level.WARNING,
```

Comment:
:ant: at least. The number of logging attempts for the ref should be limited. No need to write anything on the first cycle iterations